File: C++_20Q/Q13.cpp
#include <iostream>

void Sieve(int range) {

  int *array = new int[range];

  for (int base = 2; base < range; base++) {
    for (int multiple = base * 2; multiple < range;
         multiple = multiple + base) {
      array[multiple] = 1;
    }
  }

  for (int i = 2; i < range; i++) {
    if (array[i] == 0) {
      std::cout << i << std::endl;
    }
  }
}

int main() { Sieve(10); }

File: Java_20Q/Q13.java
public class Q13 {
  public static void Sieve(int k) {
    int[] numbers = new int[k];

    for (int base = 2; base < k; base++) {
      for (int multiple = base * 2; multiple < k; multiple += base) {

        numbers[multiple] = 1;
      }
    }

    for (int i = 2; i < k; i++){
      if(numbers[i] == 0)
        System.out.println(i);
  }

  public static void main(String[] args) {
    Sieve(50);
  }
}
